### PR TITLE
Feature/ Implement sort feature

### PR DESCRIPTION
**Pull request by JennyDoJD**
- Result:
image
image

- Ticket: [Implement Sort feature](https://trello.com/c/6tlWQv79/49-implement-sort-feature)

### CODE REVIEW

File: practice/src/js/services/baseService.js [hunk 1]
```diff
@@ -24,6 +24,34 @@ class BaseService {
...
   }
 
+  /**
+   * Sorts the items based on the specified field and order.
+   * @param {string} field - The field to sort the items by.
+   * @param {string} [orderBy=''] - The order by which to sort the items.
+   * @returns {Promise<Object[]>} - A promise resolving to an array of sorted items.
+   */
+  async sort(field, orderBy = '') {
+    const items = (await this.getAll()).slice();
```
Comment (ghost) [inline practice/src/js/services/baseService.js]: Does the mock API support sorting?

Comment (ghost) [reply practice/src/js/services/baseService.js]: I am not sure why do we need `.slice()`? can you explain?

Comment (JennyDoJD) [reply practice/src/js/services/baseService.js]: i don't want to modify the original array returned by getAll(), so i use .slice() to create a copy of it. but when i look back, i see it's redundant here, so I'll fix it

File: practice/src/js/products/productList.js [hunk 2]
```diff
@@ -59,4 +60,13 @@ export default class ProductList {
       this.productTemplate.toggleDeleteModal();
     }
   };
+
+  /**
+   * Handle sorting of products
+   * @param {string} sortOption - The selected sort option in the format "field-orderBy"
```
Comment (huyencao-agilityio) [inline practice/src/js/products/productList.js]: Update comment

File: practice/src/js/services/baseService.js [hunk 3]
```diff
@@ -11,8 +11,14 @@ class BaseService {
    * Gets data from the api
    * @returns {Promise<Object[]>} An array of data objects
    */
-  async getAll() {
-    return await this.httpClient.get(this.endpoint);
+  async getList(params = {}) {
```
Comment (huyencao-agilityio) [inline practice/src/js/services/baseService.js:15]: Update comment

**Review by huyencao-agilityio — APPROVED** [review]

**Pull request merged**